### ISSUE TITLE
Produce invalid code with collection-like initializations

### ISSUE THREAD

**Issue by AndreyZM** [bridgedotnet/Bridge#1948]
It became invalid only it case if there is a loop flow control and local variable in collection initialization exists.

If you add "if (false) continue;" into the loop in your example its will fail:
```js
Uncaught TypeError: newJObj.output is not a function
```

### Steps To Reproduce

[Deck](http://deck.net/8e5f3cd536dab281723204c495dace10)

```cs
public class AddObj : IEnumerable<KeyValuePair<string, object>>
{
    public AddObj() : base()
    {

    }

    public void Add(string key, object value)
    {
        this[key] = value;
    }

    IEnumerator<KeyValuePair<string, object>> IEnumerable<KeyValuePair<string, object>>.GetEnumerator()
    {
        throw new System.NotImplementedException();
    }

    public IEnumerator GetEnumerator()
    {
        throw new System.NotImplementedException();
    }
}

public class Test
{   
    static public void Main()
    {
        //invalid
        foreach (var item in new string[1]{"item"})
        {
            // It became invalid only it case if there is a loop flow control and local variable in collection initialization exists
            if(false)
                continue;
            
            var newJObj = new AddObj {{ "name", item }};
            System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
        
         //valid
        foreach (var item in new string[1]{"item"})
        {
            if(false)
                continue;

            var newJObj = new AddObj {{ "name", "item" }};
            System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
        
        
        //valid
        foreach (var item in new string[1]{"item"})
        {
            var newJObj = new AddObj {{ "name", item }};
            System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
    }
}
```

**Comment by LeonidVeriga** [comment]
Hi @AndreyZM,

Just improved a bit your sample to output results into Console in Deck and compare to .Net results:
```
name:5
name:1
```

Both seem to get the same results:
[Deck](http://deck.net/f57fd82cbdaca7c528b106699e456e82)
[.Net Fiddle](https://dotnetfiddle.net/V8Yehq)

Could you provide additional info on how to reproduce the issue.

**Comment by AndreyZM** [comment]
It became invalid only it case if there is a loop flow control and local variable in collection initialization exists. 
http://deck.net/b4c625a1b3fc121d2a0310aa3039fa16

```csharp
using System.Collections.Generic;
using System.Collections;

public class AddObj : IEnumerable<KeyValuePair<string, object>>
    {
		public AddObj() : base()
		{

		}

		public void Add(string key, object value)
		{
			this[key] = value;
		}

		IEnumerator<KeyValuePair<string, object>> IEnumerable<KeyValuePair<string, object>>.GetEnumerator()
		{
			throw new System.NotImplementedException();
		}

		public IEnumerator GetEnumerator()
		{
			throw new System.NotImplementedException();
		}
    }
    
public class Test
{   
    static public void Main()
    {
        //invalid
        foreach (var item in new string[1]{"item"})
        {
            if(false)
                continue;
			
           	var newJObj = new AddObj {{ "name", item }};
           	System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
        
         //valid
        foreach (var item in new string[1]{"item"})
        {
            if(false)
                continue;

           	var newJObj = new AddObj {{ "name", "item" }};
           	System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
        
        
        //valid
        foreach (var item in new string[1]{"item"})
        {
           	var newJObj = new AddObj {{ "name", item }};
           	System.Console.WriteLine(Bridge.Html5.JSON.Stringify(newJObj));
        };
    }
}
```

**Comment by AndreyZM** [comment]
If you add "if (false) continue;" into your example its will fail.

**Comment by LeonidVeriga** [comment]
@AndreyZM, thank you, it is a great catch. We will fix the issue in 15.3

**Comment by LeonidVeriga** [comment]
Updated the issue description with all new info.
